### ISSUE TITLE
Single event firing multiple transitions

### ISSUE THREAD

**Issue by ratem** [nsi-iff/fluidity#1]
In some cases a single event can fire multiple transitions in accordance to multiple guard conditions. 
For instance:
`transition(from_='request_analyzed', event='forward_analysis_result', guard = 'loan_refused', to='refusal_letter_sent')`
`transition(from_='request_analyzed', event='forward_analysis_result', guard = 'loan_accepted', to='loan_created')`

**Comment by rodrigomanhaes** [comment]
Seeing runnable code helps thinking about solutions. I have an alternative proposal:

``` python
transition(from_='request_analyzed', event='forward_analysis_result', to='loan_created', guard='loan_accepted', guard_else='refusal_letter_sent')
```

This syntax avoids repetition and ensures that guards are mutually exclusive (because there's only one). 

I'm not sure if "guard_else" is a good name, but that's the idea.

I'm not sure, part 2: conceptually, in this case we have two transitions? Or one transition being some kind of bifurcation?

**Comment by ratem** [comment]
My remarks:
1) Conceptually, there can be more than two guards/transitions for a given event
2) The mutually exclusive guards question can be solved by throwing an exception when more than one guard is true.
3) Yes, in this case we have two (or more) transitions.

**Comment by rodrigomanhaes** [comment]
Ok. Can the same event have different "froms" in different transitions?

**Comment by ratem** [comment]
Conceptually yes...
In two (or more) different states, the same event can occur, firing different transitions.

**Comment by rodrigomanhaes** [comment]
Are there something forbidden in this stateful world? :-P

**Comment by ratem** [comment]
Yep!
1) Do not obey to guards
2) A machine with only one state
;-)

**Comment by rodrigomanhaes** [comment]
closed by 3c46c2a035d909e67125f89330dc4675e226a4ca

**Comment by ratem** [comment]
Excelent!

2011/6/19 rodrigomanhaes
reply@reply.github.com:

> closed by 3c46c2a035d909e67125f89330dc4675e226a4ca
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/nsi-iff/fluidity/issues/1#issuecomment-1397282

## 

Rogerio Atem de Carvalho
